packages/flutter/lib/src/widgets/focus_manager.dart

```diff
@@ -85,26 +92,27 @@ class FocusNode extends ChangeNotifier {
   /// This method is safe to call regardless of whether this node has ever
   /// requested focus.
   void unfocus() {
-    _parent?._resignFocus(this);
-    assert(_parent == null);
+    _scopeParent?._resignFocus(this);
+    assert(_scopeParent == null);
     assert(_manager == null);
   }
 
   @override
   void dispose() {
     _manager?._willDisposeFocusNode(this);
-    _parent?._resignFocus(this);
-    assert(_parent == null);
-    assert(_manager == null);
+    _scopeParent?._resignFocus(this);
     super.dispose();
   }
 
+  @mustCallSuper
   void _notify() {
     notifyListeners();
   }
 
   @override
-  String toString() => '${describeIdentity(this)}${hasFocus ? '(FOCUSED)' : ''}';
+  String toString({DiagnosticLevel minLevel = DiagnosticLevel.debug}) {
```

Comment:
if you're implementing DiagnosticableTreeMixin you shouldn't need to implement toString

Reply:
OK.  How would I get the "(FOCUSED)" to appear after the non-deep version with the DiagnosticableTreeMixin interface?  This seemed like the way to do that, but maybe it should be a FlagProperty instead?

Reply:
@jacob314 should be able to answer that, please update the documentation to make it clear once you know the answer. :-)

Reply:
Actually I figured it out.  The answer was to us a FlagProperty, and implement Diagnosticable instead of the tree mixin (the FocusNode is only at the leaves of the tree).

Reply:
Yes FlagProperties are for exactly this sort of case. Let me know if there is anything we can do for the documentation to clarify it. For example, would it help if we clarified that you should never implement toString yourself if using DiagnnosticableTreeMixin?

Reply:
@pq Similar to the other lints you have been adding, we could add a lint to the DiagnosticsNode lints that warns if a user overrides toString directly while implementing Diagnosticable. Overriding toStirng directly indicates confusion about how the API should be used and we should push users on the right path. The lint could tell users to override debugFillProperties instead.

Reply:
Yes, that would have been a helpful lint here.

Reply:
A lint like that sounds like a win here.  Tracking here: https://github.com/dart-lang/linter/issues/1492

👍